structuralcodes/codes/mc2010/_concrete_creep_and_shrinkage.py

```diff
@@ -148,671 +183,537 @@ def _check_env_temp(T: float) -> None:
         )
 
 
-def _get_creep_shrinkage_coeffs(cem_class: str) -> dict:
-    """Get shrinkage and creep model coefficients belonging to the
-        specified cement strength class.
-
-    Defined in fib Model Code 2010 (2013), Eq. 5.1-73 and table 5.1-12.
-
-    args:
-        cem_class (str): The cement strength class that is used.
-
-    returns:
-        A dictionary with the model parameters ALPHA, ALPHA_AS,
-         ALPHA_DS1, and ALPHA_DS2.
-    """
-    _check_cem_strength_class(cem_class)
-    if cem_class.upper() in ["32.5 R", "42.5 N"]:
-        ALPHA = 0
-        ALPHA_AS = 700
-        ALPHA_DS1 = 4
-        ALPHA_DS2 = 0.012
-    elif cem_class.upper() in ["42.5 R", "52.5 N", "52.5 R"]:
-        ALPHA = 1
-        ALPHA_AS = 600
-        ALPHA_DS1 = 6
-        ALPHA_DS2 = 0.012
-    elif cem_class.upper() in ["32.5 N"]:
-        ALPHA = -1
-        ALPHA_AS = 800
-        ALPHA_DS1 = 3
-        ALPHA_DS2 = 0.013
-
-    return {
-        'alpha': ALPHA,
-        'alpha_as': ALPHA_AS,
-        'alpha_ds1': ALPHA_DS1,
-        'alpha_ds2': ALPHA_DS2,
-    }
-
-
-def _calc_temp_corr_age(t0: float, T_cur: float) -> float:
+def t_T(t0: float, T_cur: float, dt: list = None) -> float:
     """Calculate the temperature corrected concrete age in days at t0.
 
     Defined in fib Model Code 2010 (2013). Eq. 5.1-85 (only for a single
      time value input, as required in Eq. 5.1-73).
 
-    args:
+    Args:
         t0 (float): The age of the concrete in days at which the
             loading is applied.
-        T_cur (float): The temperature of the environment during curing
+        T_cur (float|list): The temperature of the environment during curing
             in degrees Celcius.
 
-    returns:
+    Keyword args:
+        dt (list): Number of days at which T_cur prevails. Required when
+            providing a list for T_cur.
+
+    Returns:
         float: The temperature corrected age of the concrete in days
             at loading.
     """
     _check_age_at_loading(t0)
-    dt = np.ones(t0)
-    T_dt = np.ones(t0) * T_cur
-    return np.sum(dt * np.exp(13.65 - (4000 / (273 + T_dt))))
-
-
-def _calc_modified_t0(t0: float, T_cur: float, TABULAR_VALUES: dict) -> float:
+    if dt is None:
+        dt = t0
+        # T_dt = T_cur
+    else:
+        try:
+            len(T_cur)
+        except TypeError as exc:
+            raise TypeError(
+                "T_cur has to be provided as list."
+            ).with_traceback(exc.__traceback__)
+        T_cur = np.array(T_cur)
+        dt = np.array(dt)
```

Comment:
Maybe this portion simplifies and improves performance if we use the method from numpy `asarray`? So we don't need to check the length before. It converts anything that can be expressed as an array to your values. Positive side note: if we already pass a numpy array, no copy is performed.
For typing, shall we stick with float|list or it is better to use ArrayLike? What do you say @mortenengen ?

Reply:
Good if we can stick to `ArrayLike` typing in order to be consistent, and great if we can avoid unecessary copies by using `.asarray()` 😃

Reply:
Done